### DISCUSSION TITLE
1 x DVB-S2 FBC - how to configure with unicable

### DISCUSSION THREAD

**Discussion by ddarek** [Barracuda09/SATPI#156] [Q&A]
Hi, 
I have a Unicable converter (1210/1420/1680/2040) connected to my DVB-S2 FBC tuner (connection a).
I have frontends 0-7.
0 and 1 are roots. 
I set:
Frontend 0 : FBC Connection A, unicable, channel freq 1210 
Frontend 2 : FBC Connection A, unicable, channel freq 1420, fbc linked
Frontend 3 : FBC Connection A, unicable, channel freq 1680, fbc linked
Frontend 4 : FBC Connection A, unicable, channel freq 2040, fbc linked

Is this config correct for unicable connected to A ??
With this configuration only 1 stream works on F0. 
What am I doing wrong?

**Comment by Barracuda09** [comment]
Hi @ddarek 

Could you give the contents of:
http://ip.of.your.satbox:8875/SatPI.xml

**Comment by ddarek** [comment]
Hi,
Sure,

In this config I tried to use 1st four tuners.
I have connected Unicable converter to first connection (A1).
I tried many combination of options. 
First adapter works, but when it tries to use another one then even this first one fails.


[SatPI.zip](https://github.com/Barracuda09/SATPI/files/8240486/SatPI.zip)

**Comment by Barracuda09** [comment]
@ddarek 

I took a quick look into the XML file but I think you missed to enter the _Channel Slot_ (UB) see:
https://github.com/Barracuda09/SATPI/wiki/Setup-Unicable-and-JESS 

So 1420 would probably be _Channel Slot_ UB: 1  (SatPI starts with UB 0)
and 1680 would be _Channel Slot_ UB: 2

Hope this will resolve the problems

**Reply by ddarek** [reply]
I have already verified it. The same result.

**Reply by Barracuda09** [reply]
Ok, but in the SatPI.xml are all channel slots set to 0.

If you try in frontend 0 -> Ch freq 1420 and UB 1 does that work on its own?

**Comment by ddarek** [comment]
It works

**Comment by Barracuda09** [comment]
Hi @ddarek 

Thanks and nice to hear your feedback. Well the UB starts in SatPI with 0 (zero) as the first SCR. This is as it is send via DiSEqc.

Usually the documentation of the LNB start at 1 (one) for the first SCR. So this can be confusing.

**Reply by ddarek** [reply]
I understand but it's not a case. I previously were using from 0.

When I enable f1,f2,f3 with settings:
Frontend -freq - slot:
F0 - 1210 - 0
F1 - 1420 - 1
F2 - 1680 - 2
F3 - 2040 - 3

All except F0 has fbclinked=true and send diseqc via root = true

I start only one stream -> works ok on F0.
When I start the second stream -> starts F1 and corrupts F0 causing to it's stop. 
After some time it starts working on F1 and F2 - works. 
I start 3rd stream and then all streams stops.

It looks like single stream works, but in case of more than 1 it stops working

Here's my current satpi.xml and logs.

[SatPI.zip](https://github.com/Barracuda09/SATPI/files/8243222/SatPI.zip)
[log.txt](https://github.com/Barracuda09/SATPI/files/8243224/log.txt)

**Reply by Barracuda09** [reply]
Which LNB are you using?

**Reply by ddarek** [reply]
I have 2 physical dvbs2 fbc tuners. 
I started from using only one device with 2 connections. 
I have connected Unicable LNB to connection A only.

**Reply by Barracuda09** [reply]
Sorry, I mean which type of Unicable LNB

**Reply by ddarek** [reply]
EN 50494
Freqs: 1210,1420,1680,2040

**Comment by Barracuda09** [comment]
But every SCR is working separate on Frontend 0?

**Reply by ddarek** [reply]
I don't understand.
But I found now that there are some problems with some muxes. Some some muxes don't work on single stream.
Strange. 
When in tvheadend I try to scan mux then I have failed.
When I switch to different sat ip server then it works.

**Reply by Barracuda09** [reply]
What I mean is that if you configure, one by one for Frontend 0 like:
Frontend - freq - slot:
  F0 - 1210 - 0 
  F0 - 1420 - 1
  F0 - 1680 - 2
  F0 - 2040 - 3

**Reply by ddarek** [reply]
Yes, as I wrote before :
Frontend -freq - slot:
F0 - 1210 - 0
F1 - 1420 - 1
F2 - 1680 - 2
F3 - 2040 - 3

**Comment by Barracuda09** [comment]
Hi @ddarek 

Could you please,  give this version a try:
[satpi_arm_1.6.2.22~g0382989_Enigma.tar.gz](https://github.com/Barracuda09/SATPI/files/8251715/satpi_arm_1.6.2.22.g0382989_Enigma.tar.gz)

**Comment by ddarek** [comment]
Hi,
It works stable now. It's ok now :)
I started 4 streams and all works ok without disrupting.

Yesterday I found also that scanning muxes don't work in tvheadend. I get "scan no data, failed".
I have in tvheadend muxes/services scanned on different box also with satpi, and some channels don't work on vuduo4kse with this satpi and previous versions.
Do you know something about problems with scanning service ?

**Reply by Barracuda09** [reply]
I have found that (my) Vu+ receiver starts sending PID data sometimes very late, at least when get these pids:

```
wget -qO- http://10.0.0.5:8875/?fe=4%26freq=11508%26sr=27500%26msys=dvbs2%26mtype=8psk%26pol=v%26fec=34%26ro=0.20%26plts=off%26pids=0,1,16,17,18
```

You could check this. It locks to the signal but data takes some time to receive.

**Reply by ddarek** [reply]
Yes, it works but after some while

**Reply by ddarek** [reply]
You're right. I have changed the timeout to 200ms and now scan was successfull

**Reply by Barracuda09** [reply]
Ok, so it locks to the channel and data is received much later in wget?

**Reply by ddarek** [reply]
Here is a log after starting this wget command:
[wget_log.txt](https://github.com/Barracuda09/SATPI/files/8252556/wget_log.txt)

**Comment by Barracuda09** [comment]
Ok, increasing the timeout int TvHeadend fixes this?

**Reply by ddarek** [reply]
Yes, on one mux I have increased value to 200 seconds and mux was scanned ok

**Reply by ddarek** [reply]
This big delay has to do something with those pids.
One service works ok, another not.

**Reply by Barracuda09** [reply]
Yes, the data rate is not big for those pids (0,1,16,17,18) only. I am still not sure were the problem exists it seems that the Vu+ is caching an amount of data in the driver, before sending it out

**Reply by ddarek** [reply]
In oscam I have :
2022/03/15 12:34:52 061DE1E5 c (dvbapi) Demuxer 0 trying to descramble PID 0 CAID 1813 PROVID 000068 ECMPID 02C4 ANY CHID PMTPID 0030 VPID 006C
2022/03/15 12:35:17 061DE1E5 c (dvbapi) Demuxer 0 stopped descrambling for program 290C (Eurosport 2 HD)

**Reply by Barracuda09** [reply]
See for info on these pids:
https://en.wikipedia.org/wiki/MPEG_transport_stream?Packet%20identifier%20(PID)#Packet_identifier_(PID)

**Comment by ddarek** [comment]
Now a lot of services don't work. Scan is also not possible. It was successful once after increaing timeout but now it don't work.
Only couple services from my provider works.
One service from mux works, but another from the same service don't.
It was better with january version. Streams were disrupting them selves (now it's ok) but all were working.
Now only couple services works, but they don't disrupt themselves.

**Reply by Barracuda09** [reply]
It this compiled with LIBDVBCSA

**Comment by ddarek** [comment]
It's a compilation you attached in this discussion thread

**Reply by Barracuda09** [reply]
Ok, I am not 100% sure anymore I used LIBDVBCSA. Do you use OSCam?

But the services that do not work, do they work in VLC or with wget?

Add the correct pids for the service:
```
wget -qO- http://10.0.0.5:8875/?fe=4%26freq=11508%26sr=27500%26msys=dvbs2%26mtype=8psk%26pol=v%26fec=34%26ro=0.20%26plts=off%26pids=0,1,16,17,18
```
For VLC use something like this in 'open network stream':
```
http://10.0.0.5:8875/?fe=4&freq=11508&sr=27500&msys=dvbs2&mtype=8psk&pol=v&fec=34&ro=0.20&plts=off&pids=0,1,16,17,18
```

**Comment by Barracuda09** [comment]
Is it something like this TvHeadend issue?
https://tvheadend.org/issues/6041

**Comment by ddarek** [comment]
I don't think it's this issue. MaxPids didn't change anything.
wget command you gave - works after long delay.
VLC not - probably because it's not using oscam. Oscam is configured to listen on 9000 port, and TVH uses it.

**Reply by Barracuda09** [reply]
Which version of TvHeadend do you use?

**Comment by Barracuda09** [comment]
Hi @ddarek 

I have done some testing and have increased the timeout to 15 sec. (See png below) 
Now it scans the muxes without premature teardown.

This is tested with `HTS Tvheadend 4.3-2010~g3b1d7a928`

![image](https://user-images.githubusercontent.com/6542445/159805700-c748a498-a3f9-4d05-aa50-076a921f4784.png)

**Comment by ddarek** [comment]
Hi, I will check it this weekend :)
I had no time recently. 
I was changing this value before. Once it even worked. I bought new LNB with 24UBs so I will have new configuration to test :)
I will check everything this weekend. 

I was using tvheadend version from feeds on openatv image. 
I will check it also with latest on seperate x64 device on debian.

**Reply by Barracuda09** [reply]
@ddarek 

Yes, I build it myself from git on x64.

But the problem is that TvHeadend is tearing down the freq. scan before the PMT pids are read from SatPI. And SatPI reads them from the Vu+ driver which takes quite some time. I have made a test program that shows that the initial scan pids (0,1,16,17,18) can take up to 7 sec to get in the buffers for reading. I am still not sure were this delay is coming from. But my test program is quite simple and straightforward with less buffering etc. and that shows the same problem.

By the way my Gigablue receiver, connected to the same setup, is taking about 3 to 4 sec.

**Reply by ddarek** [reply]
Hi, 
I have checked with latest satpi git verstion, with tvheadend 4.3-1919(2022-02-14) and the result is the same.
![image](https://user-images.githubusercontent.com/9989513/160230427-e388f396-4099-4f10-b5a1-26f68295505c.png)

**Reply by ddarek** [reply]
I have scanned all muxes I'm using with already mapped services. Couple ok, but the rest are failing.
I also think it must be some bug in vu+ drivers.

**Comment by Barracuda09** [comment]
Did you adjust the 'Timeout' and the number of inputs?

**Comment by Barracuda09** [comment]
Hi @ddarek 

I have done some digging around in the source of Tvheadend, but I think (with high probability) that there is an issue in there with parsing the PMT pids that are requested during scanning.

I am still checking my changes to the Tvheadend code to see if it is correct. I cant understand it just yet, as this piece of code seems to be there for almost 10 years. But this as, maybe going unnoticed because the 'Parent disable' is 'enabled' not so long ago.

**Comment by ddarek** [comment]
Hi, 
I hope it will help :) It would be nice to have it working.
And yes, I was increasing timeouts.

**Comment by Barracuda09** [comment]
Do you use tvheadend on x64? 

I could give you my test version

**Reply by ddarek** [reply]
Yes, I can test it :)

**Comment by Barracuda09** [comment]
@ddarek 

Here it is my version 0f the Tvheadend bin
[tvheadend.tar.gz](https://github.com/Barracuda09/SATPI/files/8358153/tvheadend.tar.gz)

**Comment by ddarek** [comment]
Unfortunately I have :  "error while loading shared libraries: libavfilter.so.6". :(
I'm using debian bullseye dist. and I have only version 7 in deps.
I will try tomorrow, thanks m8 :)

**Reply by Barracuda09** [reply]
I use here test VM with xubuntu 18.04.03

**Reply by Barracuda09** [reply]
I will give you my test version of SatPI, this evening as well.

**Comment by Barracuda09** [comment]
Here is my test version of SatPI

[satpi_arm.tar.gz](https://github.com/Barracuda09/SATPI/files/8362949/satpi_arm.tar.gz)

**Comment by ddarek** [comment]
Ok, 
I was unable to run your compilation of tvheadend. I have compiled my own compilation of tvh with newer deps.
In tvh you ment about a specific git revision or you just have local changes only ?

**Comment by Barracuda09** [comment]
No just local changes only.

**Comment by Barracuda09** [comment]
This need to be removed (or put in comment) for the test:
https://github.com/tvheadend/tvheadend/blob/3b1d7a928a8632d8c59e1fc6bb1a0a25dde9d5af/src/input/mpegts/dvb_psi_lib.c#L331-L333

**Reply by ddarek** [reply]
I have compiled tvheadend as you wrote + used your satpi binary and all muxes are failing.
Maybe I'm doing something wrong. I have spend more time on it verify it.

**Reply by Barracuda09** [reply]
You adjusted the 'Grace Period' aswell

**Comment by Barracuda09** [comment]
![Screenshot_2022-03-28_22-07-42](https://user-images.githubusercontent.com/6542445/160479201-3bbe12bb-7f06-4bcf-9396-12b05deeca1e.png)

**Comment by Barracuda09** [comment]
![Screenshot_2022-03-28_22-07-51](https://user-images.githubusercontent.com/6542445/160479258-f0a515b2-7914-4115-a419-ef7192516d98.png)

**Comment by Barracuda09** [comment]
![Screenshot_2022-03-28_22-13-59](https://user-images.githubusercontent.com/6542445/160479475-aff8d647-321e-4d9c-bb44-a08419841956.png)

**Comment by Barracuda09** [comment]
![Screenshot_2022-03-28_22-14-12](https://user-images.githubusercontent.com/6542445/160479523-b9155886-c64e-428c-a47c-e53c63307408.png)

**Comment by Barracuda09** [comment]
![Screenshot_2022-03-28_22-38-02](https://user-images.githubusercontent.com/6542445/160483065-c121c16b-ffa1-4ddc-bd47-591a48d52eea.png)

**Comment by Barracuda09** [comment]
![Screenshot_2022-03-28_22-38-13](https://user-images.githubusercontent.com/6542445/160483110-e83841c3-7ca4-423c-9e1d-d82ba907dcb2.png)

**Comment by ddarek** [comment]
Hi,
I have set tvh and satpi exactly like on your photos.
After vu+ reboot first mux scan is ok, but next ones are failing.
I attach log from satpi:
[satpi_logs.txt](https://github.com/Barracuda09/SATPI/files/8368592/satpi_logs.txt)

**Comment by ddarek** [comment]
Some later muxes scans were successful. 
Here are further logs:
[satpi_logs2.txt](https://github.com/Barracuda09/SATPI/files/8368738/satpi_logs2.txt)

And screenshot with result:
![image](https://user-images.githubusercontent.com/9989513/160549974-41fd6e15-3cdf-4ada-83b1-3209c111514d.png)

**Comment by ddarek** [comment]
Interesting is number of services.

**Reply by Barracuda09** [reply]
You can increase both the timeout to 20 sec. I had the experiance that it worked better

**Reply by Barracuda09** [reply]
For some reason it takes a long time to receive the PMT table data.

**Reply by ddarek** [reply]
Yes, I noticed it too

**Comment by Barracuda09** [comment]
Here after 2 sec it still did not receive the PMT... Strange..
```
  Tue Mar 29 08:21:14.2174 2022  7  <CRLF>
  Tue Mar 29 08:21:17.6468 2022  7  [                          src/HttpcServer.cpp:150] RTSP Stream data from client None with IP 10.0.0.2 on 
```

```
  Tue Mar 29 08:21:14.2078 2022  7  [                          src/HttpcServer.cpp:150] RTSP Stream data from client None with IP 10.0.0.2 on Port 40356: PLAY rtsp://10.0.0.5/stream=100?addpids=50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74 RTSP/1.0
  Tue Mar 29 08:21:14.2078 2022  7  Session: 0267850813
  Tue Mar 29 08:21:14.2078 2022  7  CSeq: 4
  Tue Mar 29 08:21:14.2078 2022  7  <CRLF>
  Tue Mar 29 08:21:14.2082 2022  6  [                        src/StreamManager.cpp:193] Found FrondtendID: 0 (fe=1)  StreamID: 100  SessionID: 0267850813
  Tue Mar 29 08:21:14.2084 2022  6  [                               src/Stream.cpp:207] Frontend: 0, StreamClient[0] with SessionID 0267850813
  Tue Mar 29 08:21:14.2086 2022  6  [                   src/input/dvb/Frontend.cpp:403] Frontend: 0, Parsing transport parameters...
  Tue Mar 29 08:21:14.2089 2022  7  [                   src/input/dvb/Frontend.cpp:410] Frontend: 0, Parsing transport parameters (Finished)
  Tue Mar 29 08:21:14.2091 2022  6  [                   src/input/dvb/Frontend.cpp:414] Frontend: 0, Updating frontend...
  Tue Mar 29 08:21:14.2092 2022  6  [                          src/mpegts/Filter.h:133] Frontend: 0, Updating PID filters...
  Tue Mar 29 08:21:14.2096 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0050 - PMT
  Tue Mar 29 08:21:14.2099 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0051 - PMT
  Tue Mar 29 08:21:14.2102 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0052 - PMT
  Tue Mar 29 08:21:14.2105 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0053 - PMT
  Tue Mar 29 08:21:14.2108 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0054 - PMT
  Tue Mar 29 08:21:14.2111 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0055 - PMT
  Tue Mar 29 08:21:14.2114 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0056 - PMT
  Tue Mar 29 08:21:14.2117 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0057 - PMT
  Tue Mar 29 08:21:14.2120 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0058 - PMT
  Tue Mar 29 08:21:14.2123 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0059 - PMT
  Tue Mar 29 08:21:14.2126 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0060 - PMT
  Tue Mar 29 08:21:14.2129 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0061 - PMT
  Tue Mar 29 08:21:14.2132 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0062 - PMT
  Tue Mar 29 08:21:14.2135 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0063 - PMT
  Tue Mar 29 08:21:14.2138 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0064 - PMT
  Tue Mar 29 08:21:14.2141 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0065 - PMT
  Tue Mar 29 08:21:14.2144 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0066 - PMT
  Tue Mar 29 08:21:14.2147 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0067 - PMT
  Tue Mar 29 08:21:14.2150 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0068 - PMT
  Tue Mar 29 08:21:14.2153 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0069 - PMT
  Tue Mar 29 08:21:14.2156 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0070 - PMT
  Tue Mar 29 08:21:14.2159 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0071 - PMT
  Tue Mar 29 08:21:14.2162 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0072 - PMT
  Tue Mar 29 08:21:14.2165 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0073 - PMT
  Tue Mar 29 08:21:14.2168 2022  7  [                          src/mpegts/Filter.h:158] Frontend: 0, Set filter PID: 0074 - PMT
  Tue Mar 29 08:21:14.2172 2022  6  [                   src/input/dvb/Frontend.cpp:435] Frontend: 0, Updating frontend (Finished in 7 ms)
  Tue Mar 29 08:21:14.2174 2022  7  [                          src/HttpcServer.cpp:217] Send reply in 9 ms
  Tue Mar 29 08:21:14.2174 2022  7  RTSP/1.0 200 OK
  Tue Mar 29 08:21:14.2174 2022  7  RTP-Info: url=rtsp://10.0.0.5/stream=100
  Tue Mar 29 08:21:14.2174 2022  7  CSeq: 4
  Tue Mar 29 08:21:14.2174 2022  7  Session: 0267850813
  Tue Mar 29 08:21:14.2174 2022  7  Range: npt=0.000-
  Tue Mar 29 08:21:14.2174 2022  7  <CRLF>
  Tue Mar 29 08:21:17.6468 2022  7  [                          src/HttpcServer.cpp:150] RTSP Stream data from client None with IP 10.0.0.2 on Port 40356: TEARDOWN rtsp://10.0.0.5/stream=100 RTSP/1.0
  Tue Mar 29 08:21:17.6468 2022  7  Session: 0267850813
  Tue Mar 29 08:21:17.6468 2022  7  CSeq: 5
  Tue Mar 29 08:21:17.6468 2022  7  <CRLF>
  Tue Mar 29 08:21:17.6472 2022  6  [                        src/StreamManager.cpp:193] Found FrondtendID: 0 (fe=1)  StreamID: 100  SessionID: 0267850813
  Tue Mar 29 08:21:17.6474 2022  6  [                               src/Stream.cpp:207] Frontend: 0, StreamClient[0] with SessionID 0267850813
  Tue Mar 29 08:21:17.6476 2022  6  [                               src/Stream.cpp:301] Frontend: 0, Teardown StreamClient[0] with SessionID 0267850813
  Tue Mar 29 08:21:17.8479 2022  6  [            src/output/StreamThreadRtpTcp.cpp:048] Frontend: 0, Destroy RTP/TCP stream to 10.0.0.2:40356
  Tue Mar 29 08:21:18.0483 2022  6  [           src/output/StreamThreadRtcpTcp.cpp:042] Frontend: 0, Destroy RTCP/TCP stream to 10.0.0.2:40356
  Tue Mar 29 08:21:18.1485 2022  6  [      src/decrypt/dvbapi/ClientProperties.cpp:054] Frontend: 0, Clearing OSCam filters and Keys...
  Tue Mar 29 08:21:18.1488 2022  6  [                          src/mpegts/Filter.h:113] Frontend: 0, Closing all active PID filters...
  Tue Mar 29 08:21:18.1490 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0000 - Packet Count: 000000037:000000
  Tue Mar 29 08:21:18.1493 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0001 - Packet Count: 000000185:000000
  Tue Mar 29 08:21:18.1495 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0016 - Packet Count: 000000358:000000
  Tue Mar 29 08:21:18.1503 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0017 - Packet Count: 000001467:000000
  Tue Mar 29 08:21:18.1506 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0018 - Packet Count: 000002048:000000
  Tue Mar 29 08:21:18.1508 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0050 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1510 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0051 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1512 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0052 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1514 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0053 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1516 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0054 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1518 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0055 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1520 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0056 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1522 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0057 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1524 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0058 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1526 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0059 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1528 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0060 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1530 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0061 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1532 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0062 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1534 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0063 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1536 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0064 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1539 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0065 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1541 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0066 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1543 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0067 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1545 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0068 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1547 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0069 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1549 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0070 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1551 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0071 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1553 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0072 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1555 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0073 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1562 2022  7  [                          src/mpegts/Filter.h:174] Frontend: 0, Remove filter PID: 0074 - Packet Count: 000000000:000000
  Tue Mar 29 08:21:18.1566 2022  6  [                   src/input/dvb/Frontend.cpp:646] Frontend: 0, Closing /dev/dvb/adapter0/demux0 fd: 10
  Tue Mar 29 08:21:18.1585 2022  6  [                   src/input/dvb/Frontend.cpp:631] Frontend: 0, Closing /dev/dvb/adapter0/frontend0 fd: 9
  Tue Mar 29 08:21:18.1624 2022  7  [                          src/HttpcServer.cpp:217] Send reply in 515 ms
  Tue Mar 29 08:21:18.1624 2022  7  RTSP/1.0 200 OK
  Tue Mar 29 08:21:18.1624 2022  7  CSeq: 5
  Tue Mar 29 08:21:18.1624 2022  7  Session: 0267850813
  Tue Mar 29 08:21:18.1624 2022  7  <CRLF>
```

**Comment by Barracuda09** [comment]
Did you try to decrease the PID request in Tvheadend to around 15?

**Reply by ddarek** [reply]
I did scan with longer timeouts (20sec).
I have exactly the same muxes ok.
"decrease the PID request in Tvheadend" - which setting is it ? "Maximum PIDs:" ? 
I had maximum pids = 32.

**Reply by Barracuda09** [reply]
I aswell, have found it still difficult to scan reliable with Vu+ and DVB-S2. I do not know were this delay comes from or prevent it.

**Reply by Barracuda09** [reply]
Yes 

> "Maximum PIDs:"

**Reply by ddarek** [reply]
With maximum pids=16 looks the same.

**Comment by Barracuda09** [comment]
Mmmm... strange.. I if you set timeouts to very extreem, lets say 35 sec

**Comment by ddarek** [comment]
With those settings it works the best:
![image](https://user-images.githubusercontent.com/9989513/160577192-ba9fb9f6-c3a8-41d9-bd53-bf8b6dcab170.png)
![image](https://user-images.githubusercontent.com/9989513/160577254-b8279ef6-e8f2-41cf-8a12-0cb5e90f237e.png)
![image](https://user-images.githubusercontent.com/9989513/160577318-63da2a94-8dd5-4769-8af3-d4c7909fd63f.png)
![image](https://user-images.githubusercontent.com/9989513/160577452-5bcc2df0-99b9-45e3-bdd5-ca815cc41ca3.png)

The result:
![image](https://user-images.githubusercontent.com/9989513/160577580-e7d3d72a-00b6-414b-9344-80a08695539a.png)

I think Grace period did a lot.
Also wait on tuning lock made some muxes to be ok.

**Reply by Barracuda09** [reply]
Many thanks for you support and testing

**Comment by Barracuda09** [comment]
Yes the scanning procedure is a bottle neck. I try to insert some delay's in SatPI for setting PIDs etc. to see if it improves the behaviour.

Did you test without the option `'Send full PLAY cmd'`?

**Reply by ddarek** [reply]
Not today. 
You can add some delays. I can't send more time on this today.
I have checked that some channels that was not working week ago now works.
When I increased grace period from 35 to 50 then two muxes started to work (scanning result=ok).

I found also that manual setting pilot =off on some muxes helped.

**Reply by ddarek** [reply]
I think that all those problems are caused by drivers bugs.
I have some other cheap box zgemma, with old satpi service and there were no problems.
Vu+duo4kse is much more expensive and the software has very poor quality.

**Reply by Barracuda09** [reply]
It has FBC tuners which is a nice feature. But indeed it seems a driver problem in Vu+ drivers.

But something makes me think it should perform beter.

**Reply by Barracuda09** [reply]
And maybe some setup or proc or sys file should/can be set to improve

**Comment by Barracuda09** [comment]
You should not set 'Send full PLAY cmd' it is not necessary and does not improve the reliability
